app.js

```diff
@@ -1,65 +1,54 @@
-function Star(el, count, callback) {
-    const starDiv = document.querySelector('#star');
-    const displayStar = document.querySelector('#display-star');
-    const starElements = document.querySelectorAll('#star i');
-    starElements.forEach((ele, index) => {
-        ele.setAttribute('value', index);
-    })
-    const addMouseEnter = (e) => {
-        let value = Number(e.target.getAttribute('value'));
-        starElements.forEach((ele, index) => {
-            if (index <= value) {
-                ele.classList.remove('fa-star-o');
-                ele.classList.add('fa-star');
-            }
-
-        })
-    };
-    starElements.forEach((ele) => {
-        ele.addEventListener('mouseenter', addMouseEnter);
-    });
-    const normaliseStars = (e) => {
-        starElements.forEach(ele => {
-            ele.classList.remove('fa-star');
-            ele.classList.add('fa-star-o');
-        })
+class Star {
+    constructor(el, count, callback) {
+        this.element = document.querySelector(el);
+        this.count = count;
+        this.callback = callback;
+        this.active = -1;
+        this.init();
+        this.bindEvents();
     }
-    const addClick = (e) => {
-        let value = Number(e.target.getAttribute('value'));
-        getStar(value + 1);
-        starElements.forEach((ele, index) => {
-            if (index <= value) {
-                ele.classList.remove('fa-star-o');
-                ele.classList.add('fa-star');
-            }
-        })
-        starElements.forEach(ele => {
-            ele.removeEventListener('mouseenter', addMouseEnter);
-            ele.removeEventListener('mouseleave', addMouseLeave);
-            ele.removeEventListener('click', addClick);
-        });
-        starDiv.removeEventListener('mouseleave', normaliseStars);
+    init() {
+        const fragment = document.createDocumentFragment();
+        for (let i = 1; i <= this.count; i++) {
+            const iElem = document.createElement("i");
```

Comment:
Can create a wrapper function. example getStarElement(). it will take one element and should return you the whole created element which you just append inside fragment.